### PR TITLE
Foreman endpoints for remote plumber control

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: protos/opts/ps_opts_server.proto [hunk 1]
```diff
@@ -55,4 +55,16 @@ message ServerOptions {
 
   // @gotags: kong:"help='What address to bind the built-in HTTP server to',default='0.0.0.0:9191'"
   string http_listen_address = 16;
+
+  // @gotags: kong:"help='Allow plumber to be controlled from https://console.batch.sh',env='PLUMBER_REMOTE_CONTROL_ENABLED',default=false"
```
Comment (dselans) [inline protos/opts/ps_opts_server.proto:59]: Great idea!

File: protos/opts/ps_opts_server.proto [hunk 2]
```diff
@@ -55,4 +55,16 @@ message ServerOptions {
...
+  bool remote_control_enabled = 17;
+
+  // @gotags: kong:"help='Address of Batch.sh Plumber remote control service',env='PLUMBER_REMOTE_CONTROL_ADDRESS',default='0.0.0.0:9897'"
```
Comment (dselans) [inline protos/opts/ps_opts_server.proto:62]: Should default to public address

File: protos/opts/ps_opts_server.proto [hunk 3]
```diff
@@ -55,4 +55,16 @@ message ServerOptions {
...
+  string remote_control_address = 18;
+
+  // @gotags: kong:"help='Batch.sh API token, needed to access remote control service',env='PLUMBER_REMOTE_CONTROL_API_TOKEN'"
```
Comment (dselans) [inline protos/opts/ps_opts_server.proto:65]: Dunno if it's in code but should make it part of validation - if remote control is enabled, then need to have api_token.

File: protos/opts/ps_opts_server.proto [hunk 4]
```diff
@@ -55,4 +55,16 @@ message ServerOptions {
...
+  string remote_control_api_token = 19;
+
+  // @gotags: kong:"help='Connect to remove control server without TLS',default=false"
```
Comment (dselans) [inline protos/opts/ps_opts_server.proto:68]:
```suggestion
  // @gotags: kong:"help='Connect to remote control server without TLS',default=false"
```

**Review by dselans — APPROVED** [review]

File: protos/opts/ps_opts_server.proto (hunk 3 above)
Comment (blinktag) [reply protos/opts/ps_opts_server.proto:65]: Will do!

File: protos/opts/ps_opts_server.proto (hunk 4 above)
Comment (blinktag) [reply protos/opts/ps_opts_server.proto:68]: 🤦

File: protos/opts/ps_opts_server.proto (hunk 2 above)
Comment (blinktag) [reply protos/opts/ps_opts_server.proto:62]: Fixed!

**Pull request merged**